### PR TITLE
Add cycle check in parser

### PR DESCRIPTION
**Pull request by Solpatium**

### CODE REVIEW

File: starknet_py/net/models/abi/parser.py [hunk 1]
```diff
@@ -158,3 +171,15 @@ def _group_by_entry_name(
                 )
             grouped[name] = entry
         return grouped
+
+
+def _to_json(value):
+    class DataclassSupportingEncoder(json.JSONEncoder):
+        def default(self, o):
```
Comment (war-in) [inline starknet_py/net/models/abi/parser.py:178]: Maybe rename `o` to `object`?

**Review by war-in — APPROVED** [review]
Nice and simple 🚀

File: starknet_py/net/models/abi/parser.py (hunk 1 above)
Comment (Solpatium) [reply starknet_py/net/models/abi/parser.py:178]: This is the name in parent class. If this method is called with keyword argument it would fail.

Comment (war-in) [reply starknet_py/net/models/abi/parser.py:178]: Oh, I see

**Review by cptartur — APPROVED** [review]

**Pull request merged**